python_ta/contracts/__init__.py

```diff
@@ -399,6 +404,12 @@ def _check_class_type_annotations(klass: type, instance: Any) -> None:
 
 def _check_invariants(instance, klass: type, global_scope: dict) -> None:
     """Check that the representation invariants for the instance are satisfied."""
+    if hasattr(instance, "currently_checking"):
+        # If already checking invariants for this instance, skip to avoid infinite recursion
+        return
+
+    instance.currently_checking = True
```

Comment:
Rather than using dot notation, use the `super(...).__setattr__(...)` call here. This skips the call to `new_setattr` above, so you won't need to change the implementation of that method.